src/main/java/ch/njol/skript/expressions/ExprConsumedItem.java

```diff
@@ -0,0 +1,65 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.EventRestrictedSyntax;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityShootBowEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Consumed Item")
+@Description("Represents the item consumed within an entity shoot bow event.")
+@Example("""
+	on player shoot bow:
+		if the consumed item is an arrow:
+			cancel event
+			send "You're now allowed to shoot your arrows." to shooter
+	""")
+@Since("INSERT VERSION")
+public class ExprConsumedItem extends SimpleExpression<ItemStack> implements EventRestrictedSyntax {
```

Comment:
This should probably include the consumed item in a consume item event, as well.